### PR TITLE
[FLINK-26353] flink stop --help does not list --type option

### PR DESCRIPTION
**Pull request by dawidwys**
```
Action "savepoint" triggers savepoints for a running job or disposes existing ones.

  Syntax: savepoint [OPTIONS] <Job ID> [<target directory>]
  "savepoint" action options:
     -d,--dispose <arg>       Path of savepoint to dispose.
     -j,--jarfile <jarfile>   Flink program JAR file.
     -t,--type <arg>          Describes the binary format in which a savepoint
                              should be taken. Supported options: [canonical - a
                              common format for all state backends, allow for
                              changing state backends, native = a specific
                              format for the chosen state backend, might be
                              faster to take and restore from.
```

```
Action "stop" stops a running program with a savepoint (streaming jobs only).

  Syntax: stop [OPTIONS] <Job ID>
  "stop" action options:
     -d,--drain                           Send MAX_WATERMARK before taking the
                                          savepoint and stopping the pipelne.
     -p,--savepointPath <savepointPath>   Path to the savepoint (for example
                                          hdfs:///flink/savepoint-1537). If no
                                          directory is specified, the configured
                                          default will be used
                                          ("state.savepoints.dir").
     -t,--type <arg>                      Describes the binary format in which a
                                          savepoint should be taken. Supported
                                          options: [canonical - a common format
                                          for all state backends, allow for
                                          changing state backends, native = a
                                          specific format for the chosen state
                                          backend, might be faster to take and
                                          restore from.
```

### CODE REVIEW

**Comment by dawidwys** [comment]
cc @Myasuka @rkhachatryan @pnowojski

**Comment by dawidwys** [comment]
kind reminder @knaufk @Myasuka @pnowojski

File: flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendSavepointTest.java [hunk 1]
```diff
@@ -200,7 +196,40 @@ public void testTriggerSavepointCustomFormat() throws Exception {
...
+            assertTrue(buffer.toString().contains(savepointDirectory));
+        } finally {
+            clusterClient.close();
+
+            restoreStdOutAndStdErr();
+        }
+    }
+
+    @Test
+    public void testTriggerSavepointCustomFormatLongOption() throws Exception {
```
Comment (Myasuka) [inline flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendSavepointTest.java:216]: I think the only difference of these two test methods is the option, one is `-t` and another is `--type`. We could use one base private method which contains the main body and two test case with different option key to reduce code duplication.

File: flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendStopWithSavepointTest.java [hunk 2]
```diff
@@ -133,6 +134,31 @@ public void testStopWithExplicitSavepointType() throws Exception {
...
+                    stopWithSavepointLatch.trigger();
+                    return CompletableFuture.completedFuture(savepointDirectory);
+                });
+        MockedCliFrontend testFrontend = new MockedCliFrontend(clusterClient);
+        testFrontend.stop(parameters);
+        stopWithSavepointLatch.await();
+    }
+
+    @Test
+    public void testStopWithExplicitSavepointTypeLongOption() throws Exception {
```
Comment (Myasuka) [inline flink-clients/src/test/java/org/apache/flink/client/cli/CliFrontendStopWithSavepointTest.java:147]: This is also true for these two test mthods, we can certainly avoid code duplication via introducing a base private test method with a parameter of option key.

**Review by Myasuka — COMMENTED** [review]
Thanks for creating this PR. I think we can avoid too much code duplication in tests.

**Review by Myasuka — APPROVED** [review]
LGTM.

**Pull request merged**